src/vs/workbench/parts/tasks/electron-browser/taskPanel.ts

```diff
@@ -0,0 +1,157 @@
+		@IOpenerService private openerService: IOpenerService,
+	) {
+		super(TASK_PANEL_ID, telemetryService, themeService);
+	}
+
+	public create(parent: Builder): TPromise<any> {
+		super.create(parent);
+		dom.addClass(parent.getHTMLElement(), 'task-panel');
+
+		this._builder = parent.innerHtml(getHtml());
+		const yesButton = this._builder.select('.task-panel-yes-telemetry');
+		const noButton = this._builder.select('.task-panel-no-telemetry');
+		const githubLink = this._builder.select('.task-panel-linkstyle');
+		const clickFeedback = this._builder.select('.task-panel-header-item');
+		const thanks = this._builder.select('.task-panel-thanks');
+
+		if (this.storageService.get(this.taskExperimentPart5)) {
+			clickFeedback.addClass('task-panel-hidden');
+			thanks.removeClass('task-panel-hidden');
+		}
+
+		yesButton.item(0).on('click', e => {
+			if (!this.storageService.get(this.taskExperimentPart5)) {
+				this.telemetryService.publicLog('taskPanel.yes');
+				this.storageService.store(this.taskExperimentPart5, true, StorageScope.GLOBAL);
+			}
+			clickFeedback.addClass('task-panel-hidden');
+			thanks.removeClass('task-panel-hidden');
+		});
+
+		noButton.item(0).on('click', e => {
+			if (!this.storageService.get(this.taskExperimentPart5)) {
+				this.telemetryService.publicLog('taskPanel.no');
+				this.storageService.store(this.taskExperimentPart5, true, StorageScope.GLOBAL);
+			}
+			clickFeedback.addClass('task-panel-hidden');
+			thanks.removeClass('task-panel-hidden');
+		});
+
+		githubLink.item(0).on('click', e => {
+			const node = event.target as HTMLAnchorElement;
+			if (node.href) {
+				this.openerService.open(URI.parse(node.href));
+			}
+		});
+		this._register(this.themeService.onThemeChange(theme => this._updateTheme(theme)));
+		this._updateTheme();
+
+		return TPromise.as(void 0);
+	}
+
+	public layout(dimension?: Dimension): void { }
+
+	private _updateTheme(theme?: ITheme): void {
+		const githubLink = this._builder.select('.task-panel-linkstyle');
+		const buttons = this._builder.select('.task-panel-mockup-button');
+		const taskItems = this._builder.select('.task-panel-example-item');
+		const thumbButtons = this._builder.select('.task-panel-feedback-button');
+
+		if (!theme) {
+			theme = this.themeService.getTheme();
+		}
+
+		if (theme.getColor(buttonBackground)) {
+			buttons.style('background-color', theme.getColor(buttonBackground).toString());
+		} else {
+			buttons.style('background-color', null);
+		}
+		buttons.style('color', theme.getColor(buttonForeground).toString());
+		githubLink.style('color', theme.getColor(textLinkForeground).toString());
+		taskItems.style('background-color', theme.getColor(selectBackground).toString());
+
+		if ((theme.type === DARK || theme.type === HIGH_CONTRAST) && thumbButtons.hasClass('light')) {
+			thumbButtons.removeClass('light');
+			thumbButtons.addClass('dark');
+		} else if (thumbButtons.hasClass('dark')) {
```

Comment:
Is this right? If you are switching from one dark theme to another dark theme, then it seems like it will fall into the second `if`.